es_base_block.py

```diff
@@ -1,12 +1,63 @@
+import ast
+from enum import Enum
+
 from nio.common.block.base import Block
 from nio.common.command.params.dict import DictParameter
 from nio.common.command.params.string import StringParameter
+from nio.metadata.properties import ListProperty, SelectProperty, \
+    PropertyHolder, IntProperty
 from nio.metadata.properties.expression import ExpressionProperty
 from nio.metadata.properties.string import StringProperty
 from nio.common.command import command
 from nio.common.discovery import Discoverable, DiscoverableType
 
 
+class Limitable():
+
+    """ A elasticsearch block mixin that allows you to limit results """
+
+    limit = IntProperty(title='Limit', default=0)
```

Comment:
Actually, it seems that this limit isn't functional at all. I tried putting limits of 0, 1, and 2 and it still returns all documents every time